pkg/client/auth/kubernetes/token_credentials.go

```diff
@@ -0,0 +1,55 @@
+package kubernetes
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"golang.org/x/oauth2"
+)
+
+type NativeTokenCredentials struct {
+	TokenSource oauth2.TokenSource
+}
+
+func (c *NativeTokenCredentials) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	jwt, err := c.getJWT(c.TokenSource)
+	if err != nil {
+		return nil, err
+	}
+
+	ca, err := getClusterCA()
+	if err != nil {
+		return nil, err
+	}
+
+	encodedCa := base64.RawURLEncoding.EncodeToString([]byte(ca))
+	body := fmt.Sprintf(`{"token":"%s", "ca":"%s"}`, jwt, encodedCa)
+	encoded := base64.RawURLEncoding.EncodeToString([]byte(body))
+
+	return map[string]string{
+		"authorization": "KubernetesAuth " + encoded,
+	}, nil
+}
+
+func (c *NativeTokenCredentials) RequireTransportSecurity() bool {
```

Comment:
Is this a TODO to enable at some point?

Reply:
No, you'll find this to be the same in the equivalent object in the oidc package in common.go

I believe in this case transportCredentials actually is just a euphemism for client TLS certs